pkg/sql/schemachanger/scpb/elements.proto

```diff
@@ -95,6 +96,7 @@ message ElementProto {
   IndexComment index_comment = 43 [(gogoproto.moretags) = "parent:\"PrimaryIndex, SecondaryIndex\""];
   IndexColumn index_column = 44 [(gogoproto.moretags) = "parent:\"PrimaryIndex, SecondaryIndex, TemporaryIndex, Column\""];
   IndexData index_data = 45 [(gogoproto.customname) = "IndexData", (gogoproto.moretags) = "parent:\"PrimaryIndex, SecondaryIndex, TemporaryIndex\""];
+  IndexGeoConfig index_geo_config = 46  [(gogoproto.moretags) = "parent:\"PrimaryIndex, SecondaryIndex, TemporaryIndex\""];
```

Comment:
I am curious what is the justification/benefit for introducing new elements for `geo config` and `secondary_index_partial`, instead of adding them as fields in existing elements?